IITC-Mobile: fix error that made it difficult to pick plugin file

modos189 opened the pull request:
It was impossible to select a file in a standard android file picker

modos189 committed:
Fix error that makes it impossible to select a file in a standard android file picker

johnd0e commented:
Here was only one error — wrong type `file/*`.
And I am not sure that we should change it to generic `application/*`, as filechooser filtering feature is convenient — just set proper data type.

As for 2nd change — in function `FileRequest` we may need customizable type filtering, as that function is (indirectly) available for user scripts.
May be we need data type as additional argument in this function.

johnd0e commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java:
```diff
@@ -145,7 +145,7 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                     final Intent target = new Intent(Intent.ACTION_GET_CONTENT);
                     // iitcm only parses *.user.js scripts
-                    target.setType("file/*");
+                    target.setType("application/*");
```
```suggestion
                    target.setType("application/javascript");
```

johnd0e commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java:
```diff
@@ -391,7 +391,7 @@ private FileRequest(final Uri uri) {
             // create the chooser Intent
             final Intent target = new Intent(Intent.ACTION_GET_CONTENT)
-                    .setType("text/*")
+                    .setType("application/*")
```
```suggestion
                    .setType("*/*")
```

modos189 committed:
Allow more types when choosing user.js

- text/plain: scripts from github.com
- text/javascript: from openuserjs.org / greasyfork.org
- application/octet-stream: files not from Downloads
